typed-racket-lib/typed-racket/infer/restrict.rkt

```diff
@@ -48,6 +48,14 @@
       ;; unions
       [((Union: t1s) _) (apply Un (map (λ (t1*) (restrict* t1* t2 pref resolved)) t1s))]
       [(_ (Union: t2s)) (apply Un (map (λ (t2*) (restrict* t1 t2* pref resolved)) t2s))]
+
+      ;; intersections
+      [((Intersection: t1s) _)
+       (apply -intersect (for/list ([t1 (in-set t1s)])
```

Comment:
I don't think so, since -intersect needs to briefly iterate over the arguments... it seems more sensible to just apply it to the list, right?